### PR TITLE
Instructions update - hints, feedback and avatar lightbulb/number

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR updates the following changes in the instructions area.
- adds a light color background to the hint and feedback chat bubbles,
- updates the 'lit' hint lightbulb next to the avatar in the `InstructionsCsfLeftCol` to a blue lightbulb image that is used in the hints in the 2023 HoC Dance Party progression,
- updates the number to a number circle and updates the UI.

These changes apply to all labs except Minecraft.

To add the light shaded color to hints and feedabck, I added the `backgroundColor` prop to `ChatBubble`. I used https://hexcolor.co/hex to find the appropriate lighter shaded colors of the border colors from hints and feedback.

In `Lightbulb.jsx`, I replaced the svg of the lightbulb with an updated svg from @markabarrett . I removed the 5 emitted yellow lines that are displayed when the lightbulb was 'lit' based on feedback from Mark and @samantha-code. (They are currently barely visible.) 
I updated the size and location of the lightbulb and number circle according to Mark's recommendations.

The Minecraft instructions UI is quite different from most labs, so Mark said to keep the current UI intact as it will need to go through a separate design update.

BEFORE UPDATE - Note that hint and feedback chat bubbles have white background.

When lightbulb is 'lit':

Screenshot 2023-11-16 at 4 22 20 PM

When lightbulb is not lit, i.e., all hints are displayed:

Screenshot 2023-11-16 at 4 41 26 PM

Screenshot 2023-11-17 at 10 54 07 AM

AFTER UPDATE -

When lightbulb is 'lit':

Screenshot 2023-11-17 at 10 49 01 AM

When lightbulb is not lit:

Screenshot 2023-11-17 at 10 49 17 AM

Screenshot 2023-11-17 at 10 49 29 AM

## Links
[Trello](https://trello.com/c/PnahIBmt/180-update-avatar-lightbulb-to-blue-and-update-number-to-number-circle-in-instructions)
[Slack thread](https://codedotorg.slack.com/archives/C05DMS18MEX/p1700081471226999)

## Testing story
I tested locally in the 2023 Dance HoC progression as well as non-Dance Party progressions (Poetry and Course 2). I also checked locally that Minecraft hints, feedback and hint lightbulb/numbers were not changed.
Here are screenshots from 2023 poetry progression:
| 1 | 2 | 3 |
| -- | -- | -- |
|   Screenshot 2023-11-17 at 10 45 00 AM | Screenshot 2023-11-17 at 10 44 18 AM | Screenshot 2023-11-17 at 10 44 07 AM |

As a contrast, here is Minecraft after the update - not there is NO change to hints, feedback, and the avatar lightbulb or number.
| 1 | 2 |
| -- | -- |
 | Screenshot 2023-11-17 at 11 32 40 AM | Screenshot 2023-11-17 at 11 32 53 AM |

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/Lightbulb.jsx [hunk 1]
```diff
@@ -236,13 +77,35 @@ export default class Lightbulb extends React.Component {
...
+          </g>
+        );
+        numberCircle = (
+          <g className={this.props.shouldAnimate ? 'animate-hint' : ''}>
+            <circle
+              cx="565"
+              cy="310"
+              r="125"
+              fill={color.white}
+              stroke="#dddddd"
```
Comment (levadadenys) [inline apps/src/templates/Lightbulb.jsx:103]: nit: for the context, shouldn't we use color from contants here?

File: apps/src/templates/instructions/ChatBubble.jsx [hunk 2]
```diff
@@ -57,12 +56,14 @@ const ChatBubble = ({
...
   skinId,
   borderColor,
+  backgroundColor,
   isDashed,
   ttsUrl,
   ttsMessage,
   textToSpeechEnabled,
 }) => {
   borderColor = borderColor || 'white';
+  backgroundColor = isMinecraft ? '#3B3B3B' : backgroundColor || 'white';
```
Comment (levadadenys) [inline apps/src/templates/instructions/ChatBubble.jsx:66]: nit: color constants here?

File: apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx [hunk 3]
```diff
@@ -180,6 +180,7 @@ class InstructionsCsfMiddleCol extends React.Component {
             <InlineHint
               key={hint.hintId}
               borderColor={color.light_info_500}
+              backgroundColor={'#f0ffff'}
```
Comment (levadadenys) [inline apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx:183]: also nit for these three: should we use color constants here?

Comment (levadadenys) [reply apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx:183]: or at least put these colors in the variables with some descriptive name (store them in this file only)

**Comment by bencodeorg** [comment]
I had no idea those yellow lines when the lightbulb is "lit" were there!

**Review by levadadenys — APPROVED** [review]
LGTM! That's awesome updates, thank you!
Only have couple **non blocking** questions re whether we need to store/use color constants for better description and maintenance in the future

**Review by sanchitmalhotra126 — APPROVED** [review]
+1 to Denys's comments on color constants but otherwise LGTM! great work getting this turned around in such a tight timeframe!

Seeing all the `backgroundColor` plumbing and `isMinecraft` conditional styling has me hoping we can replace this all with React context/themes at some point 🙂

File: apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx [hunk 4]
```diff
@@ -211,6 +213,7 @@ class InstructionsCsfMiddleCol extends React.Component {
         {this.props.shouldDisplayHintPrompt() && (
           <HintPrompt
             borderColor={color.light_info_500}
+            backgroundColor={'#f0ffff'}
```
Comment (bencodeorg) [inline apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx:216]: Does `HintPrompt` accept this prop?

**Comment by bencodeorg** [comment]
Just noting I think there's one other component that uses this lightbulb that wasn't mentioned here -- I think it'll look fine with the new styling (and I'm not sure tbh where you'd see it on the site), but just FYI:

https://code-dot-org.github.io/cdo-styleguide/?path=/story/dialogbuttons--try-again-with-hint

File: apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx (hunk 4 above)
Comment (fisher-alice) [reply apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx:216]: Great catch! Will add.

**Comment by fisher-alice** [comment]
> https://code-dot-org.github.io/cdo-styleguide/?path=/story/dialogbuttons--try-again-with-hint

Thanks so much for documenting this!

File: apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx (hunk 3 above)
Comment (fisher-alice) [reply apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx:183]: Thanks so much Denys! I will address these changes using constants in a fast follow-up. I am currently blockly @katiejofr from making a screencast video which includes these updates so will merge this PR including change @bencodeorg mentioned in his comment.

**Comment by fisher-alice** [comment]
> +1 to Denys's comments on color constants but otherwise LGTM! great work getting this turned around in such a tight timeframe!
> 
> Seeing all the `backgroundColor` plumbing and `isMinecraft` conditional styling has me hoping we can replace this all with React context/themes at some point 🙂

Thanks! I can add this to item (replacing with React context/themes) to the [Tech Debt doc](https://docs.google.com/document/d/1PxF26am8yamMszcEUptrfVDwl6rgZzSEr8pzuqjlu_A/edit?usp=sharing) you created!

File: apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx (hunk 3 above)
Comment (fisher-alice) [reply apps/src/templates/instructions/InstructionsCsfMiddleCol.jsx:183]: https://trello.com/c/gTzUI8Kd/182-follow-up-to-instructions-update-cleanup-refactor-of-color-constants

**Pull request merged**